packages/sanes-chrome-extension/src/routes/recovery-phrase/index.dom.spec.ts

```diff
@@ -1,43 +1,104 @@
+import TestUtils from 'react-dom/test-utils';
 import { Store } from 'redux';
-import { RootState } from '../../store/reducers';
+import * as messages from '../../extension/background/messages';
 import { aNewStore } from '../../store';
-import TestUtils from 'react-dom/test-utils';
+import { RootState } from '../../store/reducers';
 import { whenOnNavigatedToRoute } from '../../utils/test/navigation';
-import { WELCOME_ROUTE } from '../paths';
+import { randomString } from '../../utils/test/random';
+import { travelToAccount } from '../account/test/travelToAccount';
+import { ACCOUNT_STATUS_ROUTE, RECOVERY_PHRASE_ROUTE } from '../paths';
+import { handlePassPhrase, handleSecurityHint, submitAccountForm } from '../signup/test/fillSignupForm';
+import { travelToSignup } from '../signup/test/travelToSignup';
 import { travelToRecoveryPhrase } from './test/travelToRecoveryPhrase';
 
-import { withChainsDescribe } from '../../utils/test/testExecutor';
-
-withChainsDescribe('DOM > Feature > Recovery Phrase', () => {
+//NOTE Commented line below because tests were not showing for this file
+//withChainsDescribe('DOM > Feature > Recovery Phrase', () => {
+describe('DOM > Feature > Recovery Phrase', () => {
   let store: Store<RootState>;
+  let recoveryPhraseDom: React.Component;
+  let backButton: Element;
+  let exportButton: Element;
 
   beforeEach(
-    (): void => {
+    async (): Promise<void> => {
       store = aNewStore();
+      recoveryPhraseDom = await travelToRecoveryPhrase(store);
+      [backButton, exportButton] = TestUtils.scryRenderedDOMComponentsWithTag(recoveryPhraseDom, 'button');
+
+      const createPersonaResponse: messages.CreatePersonaResponse = {
+        accounts: [],
+        mnemonic: 'badge cattle stool execute involve main mirror envelope brave scrap involve simple',
+        txs: [],
+      };
+      jest.spyOn(messages, 'sendCreatePersonaMessage').mockResolvedValue(createPersonaResponse);
+
+      const getPersonaResponse: messages.GetPersonaResponse = {
+        accounts: [],
+        mnemonic: 'badge cattle stool execute involve main mirror envelope brave scrap involve simple',
+        txs: [],
+      };
+      jest.spyOn(messages, 'sendGetPersonaMessage').mockResolvedValue(getPersonaResponse);
     },
   );
 
-  it(`should contain mnemonic string and one cancel button`, async (): Promise<void> => {
-    const RecoveryPhraseDom = await travelToRecoveryPhrase(store);
+  it('Back button should exist', () => {
```

Comment:
In all the good modern testing frameworks, the method names are used to be read as part of a scentence:

```js
describe("Dog", () => {
  it("can jump", () => {
    // test that Dog can jump
  });
  it("can bark", () => {
    // test that Dog can bark
  });
});
```

is written like this to be read as: _Desribe Dog: it can jump; it can bark_.

So the texts should be choosen in a way that describe the Recovery Phrase, e.g. "[it] should have a back button"

Reply:
I was planning to do that, but I wasn't sure if the intermediate ones had to be *withChainsDescribe* or just the top one.

Reply:
`withChainsDescribe` is just a wrapper around `describe` or `xdescribe`. When `xdescribe` is used at top level, everything inside is excluded.

However, I do not suggest creating additional describe blocks. The top one is probably enough. It is more about the texts you put in the `it`s.